### ISSUE TITLE
[BUG] Printer Not Responding to M106 Command Till I Interact with TFT35 LCD

### ISSUE THREAD

**Issue by jerryfudd** [MarlinFirmware/Marlin#21864]
### Did you test the latest `bugfix-2.0.x` code?

No, but I will test it now!

### Bug Description

SKR Mini 2.0 board with TFT35 (horizontal board, not E3 variant) ignoring M106 command 'after' interacting with LCD control knob and continues to ignore until further interaction is made with LCD control knob after every M106 command sent. I rolled printer back to Marlin 2.0.7.2 and issue went away. Re-upgraded to Marlin 2.0.8 and issue returned.

I'm compiling my own Marlin for Ender 5 Plus printer. Printer is running SKR Mini 2.0 and TFT35 (horizontal board, not the square E3 variant). I'm using the Ender 3 Pro SKR Mini 2.0 configuration as a base starting point. My firmware changes over this base configuration are attached in the file Notes-E5Plus-MarlinFirmware.txt
[Notes-E5Plus-MarlinFirmware.txt](https://github.com/MarlinFirmware/Marlin/files/6452374/Notes-E5Plus-MarlinFirmware.txt)

I also have an Ender 3 Pro with the same board - an SKR Mini 2.0, except this machine uses the TFT35 E3 variant (square style variant). Again, I'm compiling my own Marlin using the Ender 3 Pro SKR Mini 2.0 example configuration file with my changes noted in the attached file Notes-E3Pro-MarlinFirmware.txt
[Notes-E3Pro-MarlinFirmware.txt](https://github.com/MarlinFirmware/Marlin/files/6452405/Notes-E3Pro-MarlinFirmware.txt)

The issue only presents on the system running the non-E3 TFT35 LCD based system when running Marlin 2.0.8, whilst I experience normal operation on the E3 TFT35 based system running Marlin 2.0.8.

The issue does not present on either printer when running Marlin 2.0.7.2.

Both the TFT35 E3 and non-E3 variants are also based upon the same BTT touchscreen firmware - just compiled for their respective model and can provide my firmware changes for these boards over the base BTT provided V.x.x.27 touchscreen firmware.

I haven't tried the Marlin 2.0.8 bugfix as yet and my printer will be tied up for the next 7hrs but will try this asap.

[Marlin 2.0.8 Affected Config Files.zip](https://github.com/MarlinFirmware/Marlin/files/6452547/Marlin.2.0.8.Affected.Config.Files.zip)

### Bug Timeline

Since Installing Marlin 2.0.8

### Expected behavior

Part fan to immediately respond to M106 S0 or M106 S200 commands

### Actual behavior

No change in part fan's operation until I pressed the TFT35 (horizontal, not E3 variant) control knob, even during a print

### Steps to Reproduce

1. Turn On Printer Running SKR Mini 2.0, TFT35 (non-E3 variant) and Marlin 2.0.8
2. Send M106 S200 command 
3. Printer responds immediately turning on the part fan
3. Use the TFT35 control knob to navigate the menu's making no changes
4. Send M106 S0 command to printer via Octoprint
5. Printer ignores command and part cooling continues to run
6. Press the TFT35 knob to access the LCD menu, part fan just turns off
7. Send the M106 S200 command to printer via Octoprint
8. Printer ignores command and part cooling fan remains off
9. Press the TFT35 knob to access LCD menu, part fan starts running
10. Downgrade to Marlin 2.0.7.2 - normal operation resumes and printer instantly responds to M106 commands at all times

### Version of Marlin Firmware

2.0.8

### Printer model

Ender 5 Plus

### Electronics

SKR Mini 2.0 & TFT35 (horizontal board not square E3 variant)

### Your Slicer

Cura

### Host Software

OctoPrint

**Comment by jerryfudd** [comment]
Ok, tried the bugfix version as of 10th May 2021 and made no difference.

Also took a videos to try and show the issue better than my explanation - both on the 2.0.8 bugfix and on 2.0.7.2.

All videos show running the M106 command to manually turn on and off the part fan

Marlin 2.0.8 Bugfix 10th May 2021 on BigTreeTech SKR Mini 2.0 with TFT35 (horizontal version, not square 'E3' version)
https://youtu.be/nyYFBDirZUo

Marlin 2.0.7.2 on same printer, as above BigTreeTech SKR Mini 2.0 with TFT35 (horizontal version, not square 'E3' version)
https://youtu.be/3j9SNQZSjY8

Marlin 2.0.8 on BigTreeTech SKR Mini 2.0 with TFT35 E3 (square version)
https://youtu.be/CKrQYUwt8qs

Whilst I accept the fact the screen form factor is the varying factor here between identical boards running near identical firmware - on the non-E3 TFT35 it does work on 2.0.7.2 and does not on 2.0.8

Likewise, we could discuss the firmware on the TFT35... however it still boils down to that it works on 2.0.7.2 and does not on 2.0.8 with no firmware changes to the screen between the two videos so something has changed which is affecting this particular screen in 2.0.8

I'd just like to add my appreciation for the great work you do also, any tone in my wording to suggest otherwise is completely unintentional - just struggling to feel I adequately and clearly explained my issue.

**Comment by jerryfudd** [comment]
it just occurred to me about a difference I've seen between the different form factor TFT35 boards and I've managed to 
narrow it further.

On the E3 TFT when making a EEPROM reset or save via the LCD it makes a nice 'beep' noise replicating how the standard Ender 3 screen sounded. This is with //#define SPEAKER in Marlin as per the default.

However, on the non-E3 TFT35, instead of a nice 'beep' noise during the same changes LCD it made a weedy triple pulse noise. I therefore uncommented the SPEAKER line in Marlin - which produced a much more pleasing noise.

It appears the side affect of uncommenting //#define SPEAKER caused the part fan issue - but only in Marlin 2.0.8. I've now commented this line back out and the problem has gone away in Marlin 2.0.8 in both the main and bugfix branches.

As a final test I uncommented  //#define SPEAKER on my E3 TFT and managed to replicate the problem onto that printer also.

**Comment by jerryfudd** [comment]
TL;DR - the short version

SKR Mini 2.0 with TFT35 (both E3 and non-E3) if //#define SPEAKER is uncommented in Marlin 2.0.8 - both stable and bugfix it causes an issue with the part fan described and as shown in the videos above where it doesn't instantly react to M106 commands till the TFT35 control knob is pressed. This issue does not present in Marlin 2.0.7.2

This also means that during prints on Marlin 2.0.8 if //#define SPEAKER is uncommented the part fan wont activate during a print also until the TFT35 control knob is pressed. This is how I first found the issue during a print as my part fan was not running when it should have been.

(side note: if anyone is curious about the 'beep' issue I was describing this is the difference between the two with speaker commented out in Marlin https://youtu.be/RvFz0uS0JHM )

**Comment by rhapsodyv** [comment]
Can you test this new envs with this PR #21488?

**Comment by jerryfudd** [comment]
> Can you test this new envs with this PR #21488?

thank you for looking into this - sorry I'm not that familiar with git..... in order to test do I just download the lastet bugfix?

**Comment by jerryfudd** [comment]
I downloaded the contents of this link https://github.com/rhapsodyv/Marlin/tree/STM32F103RC_btt_stm32 to a zip - hopefully thats correct.

will test on my Ender 3 with the SKR mini 2.0 and TFT35 E3 asap and the Ender 5 Plus with SKR mini 2.0 and TFT35 non-E3 when it has completed a print in 19hrs.

**Comment by rhapsodyv** [comment]
Yes, but you need to set this env `STM32F103RC_btt_stm32` (with `_stm32`).

**Comment by jerryfudd** [comment]
> Yes, but you need to set this env `STM32F103RC_btt_stm32` (with `_stm32`).

thank you, I'm now seeing expected behaviour from the part fan and speaker as I experienced 2.0.7.2 - which on the 'TFT35 E3' is a slightly garbled tone which is why I had always left //#define SPEAKER as commented out and why I only saw the issue on my other printer.

I'll test on the non-E3 TFT35 machine and feedback tomorrow when it finishes the current print - but based on the above I'm very positive. thank you.

As for the env I've screenshot attached for verification.
<img width="1233" alt="Screenshot 2021-05-11 at 15 01 37" src="https://user-images.githubusercontent.com/23741295/117828797-11369780-b26a-11eb-96e8-253a9d2dd0b3.png">

**Comment by jerryfudd** [comment]
> Yes, but you need to set this env `STM32F103RC_btt_stm32` (with `_stm32`).

**Comment by jerryfudd** [comment]
@rhapsodyv I was able to test this earlier than planned on my other printer and has indeed fixed the issue on my TFT35 non-E3 machine. I now get correct part fan control with //#define SPEAKER uncommented.

thank you.

**Comment by rhapsodyv** [comment]
Thanks for your test! My PR was just merged now! :-)

**Comment by jerryfudd** [comment]
> Thanks for your test! My PR was just merged now! :-)

Sorry to keep tacking onto this bug that you've resolved, but do you know the difference between the STM32F103RC_btt_stm32 and STM32F103RC_btt environments?
